### PR TITLE
Move visualization functions to `plotly` and get rid of `matplotlib` dependency

### LINKED ISSUE #348: Move visualization functions to `plotly` and get rid of `matplotlib` dependency
### Problem Description
Other libraries in the SDV ecosystem are using `plotly` to generate visuals. For example, the SDMetrics library offers [visualization utilities](https://docs.sdv.dev/sdmetrics/reports/visualization-utilities) using plotly.

However, the `Copulas` library still uses `matplotlib`. It would be nice to unify all visualizations to using `plotly`:

1. Plotly is interactive. You can zoom into specific areas, as well as toggle legend items on/off
2. We want to limit the number of external dependencies in the SDV ecosystem, as it makes it harder to maintain and forces us to make more frequent updates

### Expected behavior
See [Copulas Visualization Methods](https://sdv.dev/Copulas/api/copulas.visualization.html). Some directly correspond to SDMetrics so we can use the same graphing interface.

- `copulas.visualization.compare_1d` is equivalent to [`sdmetrics.utils.get_column_plot`](https://docs.sdv.dev/sdmetrics/reports/visualization-utilities#compare-a-synthetic-column-and-real-column-1d)
- `copulas.visualization.hist_1d` can use the same visualization as above, but it only has 1 data series (color)
- `copulas.visualization.compare_2d` is equivalent to [`sdmetrics.utils.get_colunn_pair_plot`](https://docs.sdv.dev/sdmetrics/reports/visualization-utilities#compare-a-pair-of-synthetic-columns-and-real-columns-2d)
-  `copulas.visualization.scatter_2d` can use the same visualization as above but it only has 1 data series (color)
- `copulas.visualization.compare_3d` can use the [plotly 3D scatter plots](https://plotly.com/python/3d-scatter-plots/)
- `copulas.visualization.scatter_3d` can use the same 3D scatter plot as above
- `copulas.visualization.side_by_side`: This may be a candidate for deprecation. Since plotly is interactive, there is no need to pass in extra kwargs

### Additional context
To keep things consistent, all visualization functions should also make the following changes:

1. Return the `plotly.figure` object instead of just showing it. If a user gets access to the object, they can make any modifications they like to the visualization
2. Use the same color scheme as the SDMetrics visualizations for consistency
3. Update any titles to accurately reflect what is being shown

### PR DESCRIPTION
**Pull request by fealho**
Resolve #348.

A notebook testing the new plots can be found [here](https://drive.google.com/drive/u/2/folders/1EBQXdFacfKzGsQwNbcS9dSL6vNtayWpi).

### CODE REVIEW

**Comment by frances-h** [comment]
Can you add some screenshots showing the new plots?

**Comment by fealho** [comment]
> Can you add some screenshots showing the new plots?

@frances-h Added to the description.

**Comment by amontanez24** [comment]
Task linked: [CU-860qhuhh5 Copulas - Move visualization functions to `plotly` and get rid of `matplotlib` dependency #348](https://app.clickup.com/t/860qhuhh5)

File: copulas/visualization.py [hunk 1]
```diff
@@ -1,136 +1,277 @@
...
-    ax = real.plot.scatter(x, y, color='blue', alpha=0.5, figsize=figsize)
-    ax = synth.plot.scatter(x, y, ax=ax, color='orange', alpha=0.5, figsize=figsize)
-    ax.legend(['Real', 'Synthetic'])
+    data = data.copy()
+    data['Data'] = 'Real'
 
+    return _generate_scatter_3d_plot(
+        data=data,
+        columns=columns,
+        color_discrete_map={'Real': PlotConfig.DATACEBO_DARK},
```
Comment (amontanez24) [inline copulas/visualization.py:268]: Same here

**Comment by npatki** [comment]
> > Can you add some screenshots showing the new plots?
> 
> @frances-h Added to the description.

@fealho they are not rendering for me. I think that if you upload a local ipynb notebook into Drive, then Google Colab won't render the graphics. Is it possible to create a notebook directly on Google Colab and run the code from a branch or something? Doing in Colab would be nice bc we can also verify the the interactivity that all plotly graphs come with.

**Review by npatki — CHANGES_REQUESTED** [review]
I'm leaving all my feedback here based on the notebook. 

### Plot Titles
Original issue had mentioned we should update any titles to accurately reflect what is being shown. I didn't see a title for any of the generated plots, we should add that.

I'm basing the proposed titles based off the SDMetrics visualizations, which is what we wanted to align to.

- `dist_1d`: `Data for column '<name>'`. If there is no name (eg. you're passing in a Series without a name, we could leave that out)
- `compare_1d`: `Real vs. Synthetic Data for column '<name>'` (again can leave name out if there is none)
- `scatter_2d`: `Data for columns '<name>' and '<name>'`
- `compare_2d`: `Real vs. Synthetic Data for columns '<name>' and '<name>'`
- `scatter_3d`: `Data for columns '<name>', '<name>' and '<name>'`
- `compare_3d`: `Real vs. Synthetic Data for columns '<name>', '<name>' and '<name>'`

### X and Y Axes Labels
For the 1D plots (`dist_1d` and `compare_1d`) the axes should be labeled. These can be the same as SDMetrics. (X-axis is `value` and Y-axis is `frequency`)

**Review by frances-h — APPROVED** [review]
Looks good!

**Review by npatki — APPROVED** [review]

**Comment by npatki** [comment]
Approved the title and axes changes. Though we should probably update that first notebook as a sanity check.

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**